app.py

```diff
@@ -28,18 +24,72 @@ def inject_func():
                 len=len)
 
 
-## initiate ###################
+###### before and after each request ########
+@app.before_request
+def open_conn():
+    global connections
+    conn = connections.getconn()
+    cursor = conn.cursor(cursor_factory = DictCursor)
+    g.cur = cursor
+    g.conn = conn
+    g.error = False
+
+
+@app.after_request
+def close_conn(response):
+    
+    if g.conn is not None:    
+        conn = g.conn
```

Comment:
You don't need to assign these to separate variables. Simply say `g.cur.close`.

Reply:
try to be neat :)

Reply:
It's not neat, it's cluttering the code making it longer, harder to read, and in general uglier!

Reply:
Oh, OK, I removed all of them!